app/routes/index.tsx

```diff
@@ -1,11 +1,52 @@
-import { LinksFunction } from '@remix-run/node';
+import { LinksFunction, LoaderFunction } from '@remix-run/node';
 import styles from 'react-datepicker/dist/react-datepicker.css';
 import Calendar from '~/components/Calendar';
 import Navbar from '~/components/common/navbar';
+import axios from 'axios';
+import moment from 'moment';
+import { useStore } from '~/store/useStore';
+import { useLoaderData } from '@remix-run/react';
+import { parseEvents } from '~/utils/event.utils';
+import { getEvents } from '~/constants/urls.constants';
+import { ToastContainer, toast } from 'react-toastify';
+import { useEffect } from 'react';
+
+export let loader: LoaderFunction = async ({ request }) => {
+  const cookie = request.headers.get('cookie');
+
+  const startTime = moment().subtract(1, 'months').startOf('month').unix() * 1000;
+  const endTime = moment().add(1, 'months').endOf('month').unix() * 1000;
+
+  try {
+    const response = await axios.get(getEvents(process.env.API_HOST ?? '', startTime, endTime), {
+      headers: {
+        'Content-Type': 'application/json',
+        Cookie: cookie,
+      },
+    });
+    return { events: response.data.data, error: null };
+  } catch (error) {
+    return { events: null, error: error };
+  }
+};
 
 export const links: LinksFunction = () => [{ rel: 'stylesheet', href: styles }];
 
 export default function Index() {
+  const setEvents = useStore((state) => state.setEvents);
+  const response = useLoaderData();
+
+  useEffect(() => {
+    if (response.error === null && response.events.length > 0) {
+      setEvents(parseEvents(response.events));
+    } else {
+      // TODO: redirect the user to login page on 401
+      toast.error('Unable to get events', {
```

Comment:
can move this to a global file in the future

Reply:
@isVivek99 not necessary as it is a oneliner in itself. Moving it in another file and calling from there would serve no purpose unless we are doing some additional task or providing abstraction.

Reply:
i agree that it is a one liner, in future we might want to have separate errors for different responses and it makes sense to have a constants file for this. Also we might want to move it to a root file , so that we invoke the same instance everytime instead of importing it everywhere

Reply:
@isVivek99 okay, we can decide that later when we are addressing the `TODO` part here. For now i'll resolve this conversation.